lib/experimental/interactivity-api/directives/wp-context.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Process wp-context directive attribute.
+ *
+ * @package wp-directives
+ */
+
+/**
+ * Process wp-context directive attribute.
+ *
+ * @param WP_Directive_Processor $tags Tags.
+ * @param WP_Directive_Context   $context Directive context.
+ */
+function gutenberg_interactivity_process_wp_context( $tags, $context ) {
+	if ( $tags->is_tag_closer() ) {
+		$context->rewind_context();
+		return;
+	}
+
+	$value = $tags->get_attribute( 'data-wp-context' );
+	if ( null === $value ) {
+		// No data-wp-context directive.
+		return;
+	}
+
+	$new_context = json_decode( $value, true );
+	// TODO: Error handling.
```

Comment:
it's not much work to do basic error-handling here. right now we're setting context to `null` on any JSON decode failures. maybe we should abort in this case and circle back swiftly on it

```php
$new_context = json_decode( $value, true );
if ( null === $new_context ) {
	return;
}

$context->set_context( $new_context );
```